src/gamma/main.d

```diff
@@ -239,9 +239,28 @@ void build(string[] fileNames, string outputDirectory)
 
     if (!outputDirectory.empty)
     {
+        version(Windows)
+        {
+            auto executableName = format!"%s.exe"(fileNames.front.stripExtension);
+        } 
+        else 
+        {
+            auto executableName = fileNames.front.stripExtension;
+        }
         args ~= format!"-od=%s"(outputDirectory);
-        args ~= format!"-of=%s"(fileNames.front.stripExtension);
+        args ~= format!"-of=%s"(executableName);
     }
+
+    version(Windows) 
```

Comment:
As far as I remember (the PR is from March, so I'm not completely sure anymore :-) ), this code is not  because of the user input, but because of the include files listed some lines above at https://github.com/linkrope/gamma/blob/11c93a9229e028f778573de87b4f478dda96f508/src/gamma/main.d#L238

I think, as long as we have this includes (I remember, you mentioned, we may get rid of it through a library), this is the best way of handling the different paths on Windows. What do you think?

Reply:
Replace literal string with calls to `buildPath`?
https://dlang.org/phobos/std_path.html#buildPath

    ~ buildPath("src", "io.d") ~ buildPath("src", "log.d") ~ buildPath("src", "epsilon", "soag", "listacks.d");

Reply:
Oh, thanks for the hint! That was beyond may D-knowledge, yet.

Reply:
Implemented. Please review,

Reply:
No! I mixed this up with another change. This is not implemented yet!

Reply:
BTW, is there a way to incorporate these runtime sources into the gamma executable, @linkrope?
That would make the gamma executable working standalone, compiling specified compilers to an executable as long as a DMD is installed.

Reply:
gamma should just generate a dub.json file for the generated compiler.
First problem is that there is already a dub.json file in the directory checked out:
the one for the gamma generator...

Reply:
In my VScode extension I generate the compiler source code into a configurable build directory (by default `build`, but could also be `gen` or so). Maybe we should go this way in the gamma project too? Maybe we sould always create there a subdirectory with the compilers name, in case some one has 2 EAG specifications in one project.
Then the generated compilers' _dub.json_ could be generated there.

But how to deal with the needed include files?